clang/docs/SYCLMainPageforRFC.md

```diff
@@ -0,0 +1,62 @@
+
+This design is well suited for supporting a third-party host compiler because the connection between the device and host code is standard C++ header code, which can be consumed by any conformant C++ host compiler.  However, we have observed that relying on standard C++ imposes certain limitations on the input SYCL source code.
+
+One example is the use of the C++ type that is used to “name” a kernel.  With the header file approach, these types must be forward declared in the integration header.  However, this places a limitation on these types to those that can be legally forward declared at namespace scope.  For example, it is not possible to forward declare a type that is defined at block scope or to forward declare a type that is in the “std” namespace.  Thus, these types may not be legally used to name kernels in the user’s SYCL source code.
+
+We would like to lift these limitations, at least in the common case where the user uses clang for both the host and device compilation phases.  Our thought was to implement a different integration approach for this common case, which might assume that the host compiler understands clang attributes like __builtin_sycl_unique_stable_name and decorates the LLVM IR with some new attributes.  Our thinking is very early at this point.
+
+Even if we develop this new approach, though, we think it makes sense to keep the approach based on the integration headers because this will continue to be used for a third-party host compiler.
+
+### Location for the logic for generating the integration support
+
+Another open question is the location for the logic that generates the integration headers (or the new integration information alluded to above).  This logic is currently located in the sema phase of the CFE, and the community has expressed concern about this.  We are considering two alternatives.  One option is to move the logic to the codegen phase of the CFE.  Another is to move the logic out of the CFE entirely and create a new LLVM IR pass instead.  If we do this, the new IR pass would likely run near the start of the pipeline, before other IR passes can transform the IR that is emitted by the CFE.  This option may be attractive especially if we end up supporting two integration methods because we could isolate each method in its own IR pass, and then enable one or the other pass depending on whether the user requests a third-party host compiler.
+
+### Dependency on the Khronos LLVMIR / SPIRV translator
+
+With our current approach, a SYCL source file is compiled to a fat object, which contains both host code and embedded LLVM IR for the device code.  When these objects are linked together, the LLVM IR modules from all objects are linked together, and then the sycl-post-link tool performs some final IR transformations that require visibility to the entire IR for each offload kernel.  After these final IR passes complete, we translate the LLVM IR to SPIR-V, which is embedded in the fat executable.
```

Comment:
Based on discussion during meeting, we intend to call this 'multi-targeted object, instead of 'fat object'.

Thanks